services/src/main/java/org/keycloak/services/resources/account/AccountConsole.java

```diff
@@ -128,13 +132,14 @@ public Response getMainPage() throws IOException, FreeMarkerException {
             }
             map.put("isTotpConfigured", isTotpConfigured);
 
+            map.put("deleteAccountAllowed", UserHelper.isDeleteAccountAllowed(realm, user));
```

Comment:
Remove this - checking if the user has the delete-account role should be done client-side and not added to the template.